providers/flagd/pkg/provider.go

```diff
@@ -90,11 +91,39 @@ func NewProvider(opts ...ProviderOption) *Provider {
 			CustomSyncProviderUri:   provider.providerConfiguration.CustomSyncProviderUri,
 			GrpcDialOptionsOverride: provider.providerConfiguration.GrpcDialOptionsOverride,
 		})
+	} else {
+		service = process.NewInProcessService(process.Configuration{
+			OfflineFlagSource: provider.providerConfiguration.OfflineFlagSourcePath,
+		})
 	}
 
 	provider.service = service
 
-	return provider
+	return provider, nil
+}
+
+func UpdateProvider(p *Provider) {
+	if len(p.providerConfiguration.OfflineFlagSourcePath) > 0 && p.providerConfiguration.Resolver == inProcess {
+		p.providerConfiguration.Resolver = file
+	}
+
+	if p.providerConfiguration.Port == 0 {
+		switch p.providerConfiguration.Resolver {
+		case rpc:
+			p.providerConfiguration.Port = defaultRpcPort
+		case inProcess:
+			p.providerConfiguration.Port = defaultInProcessPort
+		}
+	}
+}
+
+func CheckProvider(p *Provider) error {
```

Comment:
Similar to https://github.com/open-feature/go-sdk-contrib/pull/648/files#r1999404241, consider `validate` or something like that.